packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.types.ts

```diff
@@ -109,6 +109,12 @@ export interface IContextualMenuProps extends IBaseProps<IContextualMenu>, IWith
 
   coverTarget?: boolean;
 
+  /**
+   * If true the positioning logic will prefer flipping edges over nudging the rectangle to fit within bounds,
+   * thus making sure the the element align perfectly with target.
+   */
+  alignPerfectlyWithTarget?: boolean;
```

Comment:
Could you rework this description. Not sure if you are saying 'prefer A over B' or 'prefer flipping edges over' 
Also curious why you'd ever want the false version of this.

Reply:
This default to false which is the current behavior. What happens is instead of aligning with the target perfectly, we allow for callout to be nudged along the alignment edges. For example, below even though the directional hint is bottom left, the callout is not aligned with any edge perfectly. However, if we provide the 'alignPerfectlyWithTarget' prop, it would align either left/right or some other edge, but perfectly.

![image](https://user-images.githubusercontent.com/2162858/47676332-9ad55c80-db79-11e8-8051-c4ad961b7245.png)

Reply:
so is it basically turning 'hints' into 'restrictions'? In this mode the hint that you provide will be the limiting edge of how the callout is rendered?

We already have directionalHintFixed which locks the callout to the side (left/right/top/bottom), so this would fix the edge.

so how about `directionalHintEdgeFixed` ? Or are there some nuances that this prop doesn't cover?

Reply:
In other threads we agreed the prop should be changed to `alignTargetEdge`. Resolving this.